password/password.cabal

```diff
@@ -110,14 +134,22 @@ test-suite password-tasty
   main-is:
       Spec.hs
   other-modules:
-      Argon2
-    , Bcrypt
-    , Internal
-    , PBKDF2
-    , Scrypt
+      Internal
     , TestPolicy
     , Validate
     , Paths_password
+  if flag(argon2)
+    other-modules:
+      Argon2
+  if flag(bcrypt)
+    other-modules:
+      Bcrypt
+  if flag(pbkdf2)
+    other-modules:
+      PBKDF2
+  if flag(scrypt)
+    other-modules:
+      Scrypt
```

Comment:
Thanks for sending this PR!  This seems reasonable to me.

I don't feel super strongly either way, but I think I'd prefer to remove all of the changes in this PR for the tests.  If end-users want to disable certain algorithms, then it is okay to me if they can't run the test suite.

Or, you could possibly set the test suite as non-buildable if any of the flags is disabled.

Maybe @Vlix has a strong opinion on this?

Reply:
I can see why you'd want this.
I do have strong opinions about the implementation, but no big reservations about adding this functionality.

Reply:
@cdepillabout if you meant if I have strong opinions about adding the flags to the test suite or not, I don't see why we'd make the test-suite non-buildable. Just testing what's "activated" seems like an ok decision (even though it adds flags and CPP to the test suite 🤷 )

Reply:
I'm happy to implement this in either way, but having a subset of tests be runnable was nice for me, since it checked that the remaining algorithms still worked as expected on Mac.